File: 2105-watering-plants-ii/2105-watering-plants-ii.py
class Solution:
    def minimumRefill(self, plants: List[int], capacityA: int, capacityB: int) -> int:
        a = capacityA
        b = capacityB
        
        l = 0
        r = len(plants) - 1
        
        count = 0
        while l < r:
            if a < plants[l]:
                count += 1
                a = capacityA
            a -= plants[l]
            
            if b < plants[r]:
                count += 1
                b = capacityB
            b -= plants[r]
            
            l += 1
            r -= 1
        if l == r and a < plants[l] and b < plants[l]:
            count += 1
            
        return count
            

File: 2105-watering-plants-ii/2105-watering-plants-ii.cpp
class Solution {
public:
    int minimumRefill(vector<int>& plants, int capacityA, int capacityB) {
        int left = 0;
        int right = plants.size() - 1;
        int a = capacityA;     
        int b = capacityB;
        int count = 0;

        while (left < right){   
            if (a < plants[left]){
                count += 1;
                a = capacityA;
            }
            a -= plants[left];              
            
            if (b < plants[right]){
                count += 1;
                b = capacityB;}
             
            b -= plants[right];
            right -= 1;
            left += 1;
               
        }   
        if (left == right){
                if (a < plants[left] && b < plants[left]){
                    count += 1;                    
                }
            }   
        return count;
    }
};